routing/route/route_test.go

```diff
@@ -160,18 +160,113 @@ func TestAMPHop(t *testing.T) {
 	}
 }
 
-// TestNoForwardingParams tests packing of a hop payload without an amount or
-// expiry height.
-func TestNoForwardingParams(t *testing.T) {
+// TestBlindedHops tests packing of a hop payload for various types of hops in
+// a blinded route.
+func TestBlindedHops(t *testing.T) {
 	t.Parallel()
 
-	hop := Hop{
-		EncryptedData: []byte{1, 2, 3},
+	tests := []struct {
+		name        string
+		hop         Hop
+		nextChannel uint64
+		isFinal     bool
+		err         error
+	}{
+		{
+			name: "introduction point with next channel",
+			hop: Hop{
+				EncryptedData: []byte{1, 2, 3},
+				BlindingPoint: testPubKey,
+			},
+			nextChannel: 1,
+			isFinal:     false,
+			err:         ErrIncorrectField,
```

Comment:
Nit: `ErrIncorrectField` => `ErrOmittedField` because technically the tlv field is correct but just not allowed?

Reply:
Omitted means that we left the field out when we should have included it, which is different to the case where we included a field that shouldn't be there? Updated this to `ErrUnexpectedField`, hope that's clearer?